### ISSUE TITLE
[Pipelines] Add a T2I-Adapter pipeline

### ISSUE THREAD

**Issue by huo-ju** [huggingface/diffusers#2437]
### Model/Pipeline/Scheduler description

T2I-Adapter is a simple and small (~70M parameters, ~300M storage space) network that can provide extra guidance to pre-trained text-to-image models while freezing the original large text-to-image models.

It has [3 pre-trained models ](https://huggingface.co/TencentARC/T2I-Adapter/tree/main/models) can work with Stable Diffusion 1.4 (and other finetuned Stable Diffusion 1.4 models)

I have a basic integration [StableDiffusionT2IAdapterPipeline](https://github.com/huo-ju/diffusers/commit/4c0f1c2e1ede082078c43b78d9484e3b3be92777) , and I'm wondering if it is possible to contribute it become a official (or community) pipeline?

### Open source status

- [X] The model implementation is available
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Official implementation of T2I-Adapter: [https://github.com/TencentARC/T2I-Adapter/](https://github.com/TencentARC/T2I-Adapter/)
Pre-trained models: [https://huggingface.co/TencentARC/T2I-Adapter/tree/main/models](https://huggingface.co/TencentARC/T2I-Adapter/tree/main/models)
My basic diffusers pipeline integration: [https://github.com/huo-ju/diffusers](https://github.com/huo-ju/diffusers)

**Comment by takuma104** [comment]
Duplicated? #2390
